src/service/signup.service.ts

```diff
@@ -8,20 +8,38 @@ interface SignupData {
 }
 
 export default {
-  createUser(userdata: SignupData) {
+  async createUser(userdata: SignupData) {
     const {
       username,
       email,
       uid,
     }: { username: string; email: string; uid: string } = userdata;
 
-    // Validate username using the isSafeInput function
-    if (!validation.isSafeInput(username)) {
-      throw new Error(
-        'Invalid input. Please check your inputs.',
-      );
-    }
+    try {
+      // Validate username
+      if (!validation.isSafeInput(username) || !validation.isValidUsername(username)) {
+        throw new Error('Invalid username. Please check your input.');
+      }
+
+      // Validate email
+      if (!validation.isNotEmptyInput(email) || !validation.isValidEmail(email)) {
+        throw new Error('Invalid email. Please check your input.');
+      }
+
+      // Validate UID
+      if (!validation.isNotEmptyInput(uid) || !validation.isValidUID(uid)) {
+        throw new Error('Something went wrong. Please check your inputs.');
+      }
 
-    return modelSignup.createUser(username, email, uid);
+      return await modelSignup.createUser(username, email, uid);
+    }
+    catch (error: any) {
+      // Prisma error
+      if (error.code == "P2002") {
+        throw new Error('User with this username or email already exists.');
```

Comment:
I think there is a better to handle the errors but this should be done in an other issue.  It's okay for now but maybe create a To Do in your PR comment and add it there